source/postprocess/topography.cc

```diff
@@ -101,57 +83,196 @@ namespace aspect
-        for (unsigned int i=0; i<sizeof(columns)/sizeof(columns[0]); ++i)
-          {
-            statistics.set_precision (columns[i], 8);
-            statistics.set_scientific (columns[i], true);
-          }
+        prm.enter_subsection("Topography");
+        {
+          prm.declare_entry ("Output to file", "false",
+                             Patterns::List(Patterns::Bool()),
+                             "Whether or not to write topography to a text file named named"
+                             "'topography.NNNNN' in the output directory");
+
+          prm.declare_entry ("Time between text output", "0.",
+                             Patterns::Double (0),
+                             "The time interval between each generation of "
+                             "text output files. A value of zero indicates "
+                             "that output should be generated in each time step. "
+                             "Units: years if the "
+                             "'Use years in output instead of seconds' parameter is set; "
+                             "seconds otherwise.");
+        }
+        prm.leave_subsection();
       }
+      prm.leave_subsection();
+    }
 
-      std::ostringstream output;
-      output.precision(4);
-      output << min_topography << " m, "
-             << max_topography << " m";
 
-      return std::pair<std::string, std::string> ("Topography min/max:",
-                                                  output.str());
+    template <int dim>
+    void
+    Topography<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Topography");
+        {
+          write_to_file        = prm.get_bool ("Output to file");
+          output_interval = prm.get_double ("Time between text output");
+          if (this->convert_output_to_years())
+            output_interval *= year_in_seconds;
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
     }
+
+
+
+
   }
 }
 
 
+// explicit instantiations
 namespace aspect
 {
   namespace Postprocess
   {
     ASPECT_REGISTER_POSTPROCESSOR(Topography,
                                   "topography",
-                                  "A postprocessor intended for use with a free surface.  After every step, "
+                                  "A postprocessor intended for use with a free surface.  After every step "
                                   "it loops over all the vertices on the top surface and determines the "
                                   "maximum and minimum topography relative to a reference datum (initial "
                                   "box height for a box geometry model or initial radius for a sphere/"
-                                  "spherical shell geometry model).  Outputs topography in meters")
+                                  "If 'Topography.Output to file' is set to true, also outputs topography "
```

Comment:
This doesn't match any more -- there ought to be something after the `/` and a closing parenthesis.